reward/supply_manager.go

```diff
@@ -411,15 +411,30 @@ func (sm *supplyManager) accumulateReward(from, to uint64, fromCheckpoint *datab
 
 		// Accumulate one block
 		var (
-			header    = sm.chain.GetHeaderByNumber(num)
-			block     = sm.chain.GetBlockByNumber(num)
-			receipts  = sm.chain.GetReceiptsByBlockHash(block.Hash())
-			rules     = sm.chain.Config().Rules(new(big.Int).SetUint64(num))
-			pset, err = sm.gov.EffectiveParams(num)
+			header = sm.chain.GetHeaderByNumber(num)
+			block  = sm.chain.GetBlockByNumber(num)
+			rules  = sm.chain.Config().Rules(new(big.Int).SetUint64(num))
 		)
+		if header == nil {
+			logger.Error("Header not found", "number", num)
+			return nil, errNoBlock
+		}
+		if block == nil {
+			logger.Error("Block not found", "number", num)
+			return nil, errNoBlock
+		}
+
+		receipts := sm.chain.GetReceiptsByBlockHash(block.Hash())
+		if receipts == nil {
+			logger.Error("Receipts not found", "number", num)
+			return nil, errNoBlock
```

Comment:
Returning the same `errNoBlock` when a receipt is not found here?

Reply:
Yes I did. Both "Receipts not found" and `errNoBlock` are only printed to the log. So it's okay to return a generic errNoBlock because we're going to see "Receipts not found" anyway.